src/System.Net.Http/src/System/Net/Http/Managed/ChunkedEncodingReadStream.cs

```diff
@@ -26,7 +26,16 @@ public ChunkedEncodingReadStream(HttpConnection connection)
                 Debug.Assert(_chunkBytesRemaining == 0);
 
                 // Start of chunk, read chunk size.
-                ulong chunkSize = ParseHexSize(await _connection.ReadNextLineAsync(cancellationToken).ConfigureAwait(false));
+                ArraySegment<byte> line;
+                while (!_connection.TryReadNextLine(out line))
+                {
+                    if (!await _connection.FillAsync(cancellationToken).ConfigureAwait(false))
+                    {
+                        throw new IOException(SR.net_http_invalid_response);
```

Comment:
Or, if every time we call FillAsync we expect it to successfully get more or else it's an error, the exception can just be moved into FillAsync.